
Description: retry forever after mouse is dettached, since you never know
 when it can be attached back.
Status: sent upstream

--- src.old/usr.sbin/moused/moused.c	2005-02-20 16:47:05.000000000 +0100
+++ src/usr.sbin/moused/moused.c	2005-02-20 16:46:25.000000000 +0100
@@ -778,9 +778,9 @@
 	usage();
     }
 
-    retry = 1;
+    retry = 0;
     if (strncmp(rodent.portname, "/dev/ums", 8) == 0) {
-	if (usbmodule() != 0)
+	if ((retry != 0) && (usbmodule() != 0))
 	    retry = 5;
     }
 
@@ -790,7 +790,7 @@
 	    signal(SIGINT , cleanup);
 	    signal(SIGQUIT, cleanup);
 	    signal(SIGTERM, cleanup);
-	    for (i = 0; i < retry; ++i) {
+	    for (i = 0; i < retry || retry == 0; ++i) {
 		if (i > 0)
 		    sleep(2);
 		rodent.mfd = open(rodent.portname, O_RDWR | O_NONBLOCK);
